workers/unity/Packages/com.improbable.gdk.tools/RedirectedProcess.cs

```diff
@@ -95,6 +104,89 @@ void OnReceived(object sender, DataReceivedEventArgs args)
             }
         }
 
+        /// <summary>
+        ///     Runs the redirected process and returns a task which can be waited on.
+        /// </summary>
+        /// <param name="workingDirectory">The directory to run the command from.</param>
+        /// <param name="command">The filename to run.</param>
+        /// <param name="arguments">Parameters that will be passed to the command.</param>
+        /// <param name="redirectStdout">Redirect standard output to Debug.Log</param>
+        /// <param name="redirectStderr">Redirect standard error to Debug.LogError</param>
+        /// <returns>A task which would return the exit code and output.</returns>
+        public static async Task<RedirectedProcessResult> RunInAsync(string workingDirectory, string command, string[] arguments, bool redirectStdout = false, bool redirectStderr = false)
+        {
+            var info = new ProcessStartInfo(command, string.Join(" ", arguments))
+            {
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                WorkingDirectory = workingDirectory
+            };
+
+            return await Task.Run(() =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    if (process == null)
+                    {
+                        throw new Exception(
+                            $"Failed to run {info.FileName} {info.Arguments}\nIs the .NET Core SDK installed?");
+                    }
+
+                    process.EnableRaisingEvents = true;
+
+                    var processStandardOutput = new List<string>();
+                    var processStandardError = new List<string>();
+
+                    void OnStandardOutput(object sender, DataReceivedEventArgs args)
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            if (redirectStdout)
+                            {
+                                Debug.Log(args.Data);
+                            }
+                            lock (processStandardOutput)
+                            {
+                                processStandardOutput.Add(args.Data);
+                            }
+                        }
+                    }
+
+                    void OnStandardError(object sender, DataReceivedEventArgs args)
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            if (redirectStderr)
+                            {
+                                Debug.LogError(args.Data);
+                            }
+                            lock (processStandardOutput)
```

Comment:
Can also set the process sync object to do this. I think I mildly prefer the lock though.